### PR TITLE
Associate Nyaovim with common text files in OSX

### PR DESCRIPTION
**Pull request by haifengkao**
### What was a problem?
Link Nyaovim to common text files in OSX.

### How this PR fixes the problem?
Extend Info.plist to cover common text files
The file extension list is copied from sublime text

Note this PR covers https://github.com/rhysd/NyaoVim/pull/120
Therefore,  https://github.com/rhysd/NyaoVim/pull/120 should not be used if this PR is accepted. 

### Check lists (check `x` in `[ ]` of list items)

- [x] Coding style (if any code was modified)
- [x] Confirmed
  - OS: OSX 10.13
  - `nvim` version: NVIM v0.2.0-1303-gdd391bfc

### Additional Comments (if any)
love you~~~

### CODE REVIEW

**Comment by rhysd** [comment]
I'm not releasing OS X package with this command yet, but it looks good. Thank you for the update and I apologize late response.

**Comment by rhysd** [comment]
@haifengkao Could you merge the upstream master branch? I think CI will pass by it.

**Comment by haifengkao** [comment]
I did. Hope that it will pass the CI

File: resources/osx_plist/file_associations.plist [hunk 1]
```diff
@@ -0,0 +1,809 @@
...
+			<string>JSON</string>
+			<key>CFBundleTypeExtensions</key>
+			<array>
+				<string>json</string>
+				<string>sublime-settings</string>
+				<string>sublime-menu</string>
+				<string>sublime-keymap</string>
+				<string>sublime-theme</string>
+				<string>sublime-build</string>
+				<string>sublime-project</string>
```
Comment (rhysd) [inline resources/osx_plist/file_associations.plist]: Should 'sublime-*' remain?

**Comment by rhysd** [comment]
Hm, I don't know why CI fails... Installing 'neovim' package via PPA does not work.

**Comment by haifengkao** [comment]
I guess it's fine to remove sublime related extension

**Comment by rhysd** [comment]
I missed important point.

To include the file from SublimeText, we need to clarify it's ok in terms of license of Sublime text. This PR does not contain any license description. And Sublime text is a proprietary software. I want to ensure that including the file from Sublime text does not cause any problem.

**Comment by haifengkao** [comment]
hmm...
I don't think this file is protected by the copyright law. The law only protects the content that shows creativeness. This file is a standard form required by apple. I can write my own form, but the result is pretty much the same.

**Comment by rhysd** [comment]
um, I agree that this feature is awesome, but it's not acceptable to include a file whose license is unclear. How about Atom or VS Code?

**Comment by haifengkao** [comment]
atom's file is here https://github.com/atom/atom/blob/master/resources/mac/atom-Info.plist
They looks pretty much the same. 
I guess I need to teach githubbers the correct copyright knowledge.

**Comment by rhysd** [comment]
I think we can use Atom's plist file instead because Atom is licensed with MIT license. We can copy the file, and write [the license description (copyright notice)](https://github.com/atom/atom/blob/master/LICENSE.md) as comment in it.

**Comment by haifengkao** [comment]
[Copyrightability of Tables, Charts, and Graphs](https://deepblue.lib.umich.edu/bitstream/handle/2027.42/83329/copyrightability_of_tables_charts_and_graphs.pdf)  can give you some info about copyright laws.

**Comment by haifengkao** [comment]
as you requested.

**Comment by haifengkao** [comment]
I guess it's very hard to change people's mind with the facts.

**Pull request merged**